uspv/msghandler.go

```diff
@@ -13,7 +13,11 @@ func (s *SPVCon) incomingMessageHandler() {
 		n, xm, _, err := wire.ReadMessageWithEncodingN(s.con, s.localVersion,
 			wire.BitcoinNet(s.Param.NetMagicBytes), wire.LatestEncoding)
 		if err != nil {
-			log.Printf("ReadMessageWithEncodingN error.  Disconnecting: %s\n", err.Error())
+			log.Printf("ReadMessageWithEncodingN error.  Disconnecting from given peer. %s\n", err.Error())
+			if s.randomNodesOK { // if user wants to connect to localhost, let him do so
+				s.Connect("yes") // really any YupString here
+			}
+			s.con.Close()
```

Comment:
this will connect then immediately close the connection if s.randomNodesOK is true

Reply:
noob mistake, sorry!

Reply:
No prob... still doesn't work though.  Now it quits the message handler after it re-connects.